src/components/structures/RoomView.js

```diff
@@ -1140,12 +1141,116 @@ module.exports = createReactClass({
         }
 
         debuglog("sending search request");
+        const platform = PlatformPeg.get();
```

Comment:
These bits would also be better off in some kind of "searching" module so that here we're just calling search effectively and the change will approximately be switching to the new module instead of the client directly.